nixos/modules/services/printing/lprint.nix

```diff
@@ -0,0 +1,520 @@
+        '';
+      };
+      top = lib.mkOption {
+        type = types.int;
+        description = ''
+          The top margin.
+        '';
+      };
+      left = lib.mkOption {
+        type = types.int;
+        description = ''
+          The left margin.
+        '';
+      };
+      right = lib.mkOption {
+        type = types.int;
+        description = ''
+          The right margin.
+        '';
+      };
+      width = lib.mkOption {
+        type = types.int;
+        description = ''
+          The width of the media.
+        '';
+      };
+      length = lib.mkOption {
+        type = types.int;
+        description = ''
+          The length of the media.
+        '';
+      };
+      source = lib.mkOption {
+        type = types.str;
+        description = ''
+          The source of the media.
+        '';
+      };
+      tracking = lib.mkOption {
+        type = types.str;
+        description = ''
+          The tracking of the media.
+        '';
+      };
+      type = lib.mkOption {
+        type = types.str;
+        description = ''
+          The type of the media.
+        '';
+      };
+    };
+  };
+  printerPapplConfigType = lib.mkMerge [
+    papplConfigType
+    (types.submodule {
+      options = {
+        maxActiveJobs = lib.mkOption {
+          type = types.nullOr types.int;
+          default = null;
+          description = ''
+            The maximum number of active jobs.
+          '';
+        };
+        maxCompletedJobs = lib.mkOption {
+          type = types.nullOr types.int;
+          default = null;
+          description = ''
+            The maximum number of completed jobs.
+          '';
+        };
+        defaultMediaCollection = lib.mkOption {
+          type = mediaCollectionType;
+          description = ''
+            The default media collection.
+          '';
+        };
+        additionalMediaCollections = lib.mkOption {
+          type = types.list mediaCollectionType;
+          default = [];
+          description = ''
+            Additional media collections.
+          '';
+        };
+        defaultOrientation = lib.mkOption {
+          type = types.nullOr (types.oneOf [ types.enum [
+            "portrait"
+            "landscape"
+            "reverse-landscape"
+            "reverse-portrait"
+            "none"
+          ] types.str ]);
+          default = "none";
+          description = ''
+            The default orientation.
+          '';
+        };
+        defaultColorMode = lib.mkOption {
+          type = types.oneOf [ types.enum [
+            "auto"
+            "color"
+            "monochrome"
+          ] types.str ];
+          default = "auto";
+          description = ''
+            The default color mode.
+          '';
+        };
+        defaultContentOptimize = lib.mkOption {
+          type = types.oneOf [ types.enum [
+            "auto"
+            "graphic"
+            "photo"
+            "text"
+          ] types.str ];
```

Comment:
`oneOf [ types.enum [<whatever>] types.str]` is pointless

Reply:
Those are left in from testing. Once I've double checked that the enum values are correct, I'll remove the `oneOf`.